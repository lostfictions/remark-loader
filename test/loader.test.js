/* global describe, expect, it */
const { getCompiler, compile } = require("./compiler");
const execute = require("./execute");

describe("remark-loader", () => {
  it("handles a basic file with attributes", async () => {
    const compiler = getCompiler("./fixtures/basic.js");
    const stats = await compile(compiler);
    const result = execute("bundle.js", compiler, stats);

    expect(result).toMatchSnapshot();
    expect(result.attributes).toEqual({ title: "my good post" });
  });

  it("handles a file with complex attributes", async () => {
    const compiler = getCompiler("./fixtures/complex-attributes.js");
    const stats = await compile(compiler);
    const result = execute("bundle.js", compiler, stats);

    expect(result).toMatchSnapshot();
  });

  it("handles a file without frontmatter", async () => {
    const compiler = getCompiler("./fixtures/no-frontmatter.js");
    const stats = await compile(compiler);
    const result = execute("bundle.js", compiler, stats);

    expect(result.attributes).toEqual({});
    expect(result.default).toMatchSnapshot();
  });

  it("handles an empty file", async () => {
    const compiler = getCompiler("./fixtures/empty.js");
    const stats = await compile(compiler);
    const result = execute("bundle.js", compiler, stats);

    expect(result.attributes).toEqual({});
    expect(result.default.trim()).toEqual("");
  });
});
