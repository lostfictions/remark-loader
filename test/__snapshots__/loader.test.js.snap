// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-loader handles a basic file with attributes 1`] = `
Object {
  "attributes": Object {
    "title": "my good post",
  },
  "default": "<h1>hello world!</h1>
<p>a fine post here</p>
",
  "plaintext": "hello world!

a fine post here
",
}
`;

exports[`remark-loader handles a file with complex attributes 1`] = `
Object {
  "attributes": Object {
    "info": Object {
      "data": "yes",
      "list": Array [
        "a thing",
        "another thing",
        "a third thing",
        Object {
          "nested object": Object {
            "it's": "happening",
            "why": "why not",
          },
        },
      ],
      "time": "soon",
    },
    "title": "a fine post",
  },
  "default": "<h1>hello world!</h1>
<p>here's my good content</p>
",
  "plaintext": "hello world!

here's my good content
",
}
`;

exports[`remark-loader handles a file without frontmatter 1`] = `
"<h1>hello world!</h1>
<p>another great post</p>
"
`;
